src/nip.py

```diff
@@ -131,28 +131,18 @@ def handle_self(self, name):
 
     def handle_subdomains(self, qname):
         subdomain = qname[0:qname.find(self.domain) - 1]
-        # Split foo.bar.10-0-0-1 in parts
-        subdomain_parts = subdomain.split('.')
+        match = re.findall('^(?:.+\.)?(\d{1,3}[-.]\d{1,3}[-.]\d{1,3}[-.]\d{1,3})$', subdomain)
 
-        # Take the last part as this is the IP separated with dashes
-        ip_dashes = subdomain_parts[-1]
-        subparts = ip_dashes.split('-')
-
-        if len(subparts) < 4:
+        if not match:
             if DEBUG:
-                log('subparts less than 4')
+                log('%s is invalid format' % subdomain)
             self.handle_self(qname)
             return
 
-        ipaddress = subparts[-4:]
+        ipaddress = re.split('[-.]', match[0])
         if DEBUG:
             log('ip: %s' % ipaddress)
         for part in ipaddress:
-            if re.match('^\d{1,3}$', part) is None:
-                if DEBUG:
-                    log('%s is not a number' % part)
-                self.handle_self(qname)
-                return
```

Comment:
I would like to keep this code

Reply:
@resmo 
I got it. I reverted this change in 751ccb5.